087-external-certificate-manager.md

```diff
@@ -0,0 +1,241 @@
+4. Roll the Kafka pods to use the new Kafka certificates, updating the cert generation annotations on the Pods.
+5. Then on the next reconciliation, since the Kafka pods now have correct cert and key generation, copy over the new operator certificate.
+6. Since the Kafka pods now have correct cert generation, remove the old CA public cert from the `<CLUSTER_NAME>-cluster-ca-cert` Secret
+
+I have created diagrams for the existing flow and the flow when using cert-manager for [certificate renewals](./images/087-cert-renewals.png) and the flow for when cert-manager issues [new end-entity certificates](./images/087-new-ee-certs.png).
+
+### User Operator/Clients CA
+TODO
+
+## Affected/not affected projects
+
+This affects the Cluster Operator and User Operator.
+
+## Compatibility
+
+This feature will be optional and disabled by default.
+
+### Migrating to this feature
+
+To start using this feature in an existing Kafka cluster the user must:
+1. Install cert-manager and create an `Issuer`.
+2. Create Secrets to store the Cluster CA and/or Clients CA public certs
+3. Update the `Kafka` resource to have `clusterCa.type` and/or `clientsCa.type` set to `cert-manager.io`, and `certManagerIssuerRef` and `publicCert` configured.
+4. Resume reconciliation.
+
+When using this feature for the ClientsCa:
+* On the next Cluster operator reconciliation Strimzi will copy the new certificate over to `<CLUSTER_NAME>-clients-ca-cert` (renaming and keeping the old certificate), increment the cert generation annotation and add the key generation annotation.
+* Strimzi will roll the Kafka pods to trust the new CA cert.
+* On the next User Operator reconciliation Strimzi will create `Certificate` resources for all the existing `KafkaUser` custom resources.
+
+When using this feature for the ClusterCa:
+* On the next Cluster operator reconciliation Strimzi will copy the new certificate over to `<CLUSTER_NAME>-cluster-ca-cert` (renaming and keeping the old certificate), increment the cert generation annotation and add the key generation annotation.
+* Strimzi will roll the pods once to trust the new CA cert.
+* Strimzi will create `Certificate` resources for all the components and wait for the certificates to be issued.
+* Strimzi will copy over the new certificates into the pod Secrets.
+* Strimzi will roll the pods to use the new certificates.
+
+Once all the pods have the correct cert generation annotation Strimzi can update the `<CLUSTER_NAME>-cluster-ca-cert` and/or `<CLUSTER_NAME>-clients-ca-cert` Secrets to remove the old CA cert and delete the `<CLUSTER_NAME>-cluster-ca` and/or `<CLUSTER_NAME>-clients-ca` Secrets.
+
+### Stopping using this feature
+
+To revert to user managed CAs the user must:
+1. Pause reconciliation for their Kafka cluster.
+2. Update the `<CLUSTER_NAME>-cluster-ca-cert` and/or `<CLUSTER_NAME>-clients-ca-cert` Secrets to:
+   1. contain their public CA cert (keeping the old cert-manager one)
```

Comment:
what does it exactly mean? Are we moving to use Strimzi to generate the CAs right? What's "their public CA cert"?

Reply:
This is moving to user managed CAs. So where the user provides the CA public and private keys

Reply:
Ah right the section clearly is "To revert to user managed CAs the user must" :-D